base/iterators.jl

```diff
@@ -1095,4 +1094,35 @@ eltype(::Type{Stateful{T, VS}} where VS) where {T} = eltype(T)
 IteratorEltype(::Type{Stateful{T,VS}}) where {T,VS} = IteratorEltype(T)
 length(s::Stateful) = length(s.itr) - s.taken
 
+"""
+    only(x)
+
+Returns the one and only element of collection `x`, and throws an `ArgumentError` if the
+collection has zero or multiple elements.
+"""
+Base.@propagate_inbounds function only(x)
+    i = iterate(x)
+    @boundscheck if i === nothing
+        throw(ArgumentError("Collection is empty, must contain exactly 1 element"))
+    end
+    (ret, state) = i
+    j = iterate(x, state)
+    @boundscheck if j !== nothing
```

Comment:
```suggestion
    @boundscheck if iterate(x,state) !== nothing
```

So that if we are `@inbounds` we don't call `iterate`

Reply:
thanks